Source/Core/Core/IOS/FS/FileSystem.cpp

```diff
@@ -72,6 +72,20 @@ void FileSystem::Init()
     CreateDirectory(0, 0, "/tmp", 0, Mode::ReadWrite, Mode::ReadWrite, Mode::ReadWrite);
 }
 
+Result<FileHandle> FileSystem::CreateAndOpenFile(Uid uid, Gid gid, const std::string& path,
+                                                 Mode owner_mode, Mode group_mode, Mode other_mode)
+{
+  Result<FileHandle> file = OpenFile(uid, gid, path, Mode::ReadWrite);
```

Comment:
I think using ReadWrite here makes sense, because you'll almost always want to write to a file after creating it. Would it be better to make it a parameter (perhaps with default = Mode::ReadWrite) so this can be overridden if necessary?

I used the same enum because the exact same values are stored in [the filesystem](https://wiibrew.org/wiki/NAND) in the file access mode bit with the same meaning.

Reply:
Not really sure, hence why I wrote it like a question instead of a request. Some places used to use `Mode::Write` before (which probably doesn't make much difference).
I guess we can always go and add the parameter later if we need it.